pkg/yaml/yaml.go

```diff
@@ -146,3 +148,37 @@ func findScalarNode(node *yaml.Node, keyPath []string) (int, int, error) {
 	}
 	return 0, 0, fmt.Errorf("key path not found")
 }
+
+// MergeFiles merges a list of YAML files.
+func MergeFiles(inputPaths []string, outputPath string) error {
+	if len(inputPaths) == 0 || outputPath == "" {
+		return fmt.Errorf("inFiles and OutFile must not be empty")
+	}
+
+	// read first YAML file
+	mergedNode, err := kyaml.ReadFile(inputPaths[0])
+	if err != nil {
+		return fmt.Errorf("error parsing first input file: %w", err)
+	}
+
+	// read all other YAML file and apply the patch
+	for i := 1; i < len(inputPaths); i++ {
+		patchNode, fileErr := kyaml.ReadFile(inputPaths[i])
+		if fileErr != nil {
+			return fmt.Errorf("error parsing input file %s: %w", inputPaths[i], fileErr)
+		}
+
+		mergedNode, err = merge2.Merge(patchNode, mergedNode, kyaml.MergeOptions{ListIncreaseDirection: 1})
```

Comment:
`ListIncreaseDirection` is a magic number I feel should be documented/comment or use a named constant